app/services/api/beta/search_result_statistics_service.rb

```diff
@@ -0,0 +1,91 @@
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['description'] ||= goods_nomenclature_hit.chapter_description
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['score'] ||= 0
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['cnt'] ||= 0
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['cnt'] += 1
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['score'] += goods_nomenclature_hit.score
+          @chapter_statistics[goods_nomenclature_hit.chapter_id]['avg'] ||= mean_average_chapter_score_for(goods_nomenclature_hit.chapter_id)
+        end
+
+        @chapter_statistics
+      end
+
+      def generate_heading_statistics
+        accumulations = 0
+        goods_nomenclature_hits_with_headings.each do |goods_nomenclature_hit|
+          accumulations += 1
+          @heading_statistics[goods_nomenclature_hit.heading_id] ||= Hashie::TariffMash.new
+          @heading_statistics[goods_nomenclature_hit.heading_id]['id'] ||= goods_nomenclature_hit.heading_id
+          @heading_statistics[goods_nomenclature_hit.heading_id]['description'] ||= goods_nomenclature_hit.heading_description
+          @heading_statistics[goods_nomenclature_hit.heading_id]['chapter_id'] ||= goods_nomenclature_hit.chapter_id
+          @heading_statistics[goods_nomenclature_hit.heading_id]['chapter_description'] ||= goods_nomenclature_hit.chapter_description
+          @heading_statistics[goods_nomenclature_hit.heading_id]['score'] ||= 0
+          @heading_statistics[goods_nomenclature_hit.heading_id]['score'] += goods_nomenclature_hit.score
+          @heading_statistics[goods_nomenclature_hit.heading_id]['cnt'] ||= 0
+          @heading_statistics[goods_nomenclature_hit.heading_id]['cnt'] += 1
+          @heading_statistics[goods_nomenclature_hit.heading_id]['avg'] ||= mean_average_heading_score_for(goods_nomenclature_hit.heading_id)
+          @heading_statistics[goods_nomenclature_hit.heading_id]['chapter_score'] ||= @chapter_statistics[goods_nomenclature_hit.chapter_id]['score']
+        end
+
+        @heading_statistics
+      end
+
+      def goods_nomenclature_hits_with_headings
+        goods_nomenclature_hits.reject do |goods_nomenclature_hit|
+          goods_nomenclature_hit.goods_nomenclature_class.chapter?
+        end
+      end
+
+      def mean_average_heading_score_for(heading_id)
+        mean_average_scores do |goods_nomenclature_hit|
+          goods_nomenclature_hit.heading_id == heading_id
+        end
+      end
+
+      def mean_average_chapter_score_for(chapter_id)
+        mean_average_scores do |goods_nomenclature_hit|
+          goods_nomenclature_hit.chapter_id == chapter_id
+        end
+      end
+
+      def mean_average_scores
+        scores = []
+        count = 0
+
+        goods_nomenclature_hits_with_headings.each do |goods_nomenclature_hit|
+          if yield goods_nomenclature_hit
```

Comment:
it's not clear to me what the if is actually checking, maybe that 
```
if  yield(goods_nomenclature_hit) 
```
is not nil? (or present?)

Reply:
Well, the result of yield is always a boolean. So equivalent would be
```ruby
 if (yield goods_nomenclature_hit == true)
```